NetKet/GroundState/py_exact.hpp

```diff
@@ -64,11 +64,11 @@ void AddExactModule(py::module &m) {
                >>> L = 20
                >>> graph = nk.graph.Hypercube(L, n_dim=1, pbc=True)
                >>> hilbert = nk.hilbert.Spin(graph, 0.5)
-               >>> idx = nk.hilbert.HilbertIndex(hilbert)
+               >>> n_states = hilbert.index.n_states
```

Comment:
honestly, at this point I would go for removing completely the index object and make those just members of `hilbert`. Would there be an issue to do that? Especially if the space cannot be indexed?

Reply:
Sounds good. I did this for the Python interface. The `HilbertIndex` object still exists in C++, because I think it is nice to have all the methods that only work for indexable spaces in one place.